examples/Phonon_Caustics/src/Caustic_PhononDetectorConstruction.cc

```diff
@@ -0,0 +1,199 @@
+#include "G4SystemOfUnits.hh"
+#include "G4TransportationManager.hh"
+#include "G4Tubs.hh"
+#include "G4UserLimits.hh"
+#include "G4VisAttributes.hh"
+#include "Caustic_PhononConfigManager.hh"
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo....
+
+Caustic_PhononDetectorConstruction::Caustic_PhononDetectorConstruction()
+  : fLiquidHelium(0),fBolometer(0),fOxigen(0),fSubstrate(0),
+    fWorldPhys(0), topSurfProp(0), wallSurfProp(0),
+    electrodeSensitivity(0), fConstructed(false) {;}
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo....
+
+Caustic_PhononDetectorConstruction::~Caustic_PhononDetectorConstruction() {
+  delete topSurfProp;
+
+  delete wallSurfProp;
+}
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo....
+
+G4VPhysicalVolume* Caustic_PhononDetectorConstruction::Construct()
+{
+  if (fConstructed) {
+    if (!G4RunManager::IfGeometryHasBeenDestroyed()) {
+      // Run manager hasn't cleaned volume stores. This code shouldn't execute
+      G4GeometryManager::GetInstance()->OpenGeometry();
+      G4PhysicalVolumeStore::GetInstance()->Clean();
+      G4LogicalVolumeStore::GetInstance()->Clean();
+      G4SolidStore::GetInstance()->Clean();
+    }
+    // Have to completely remove all lattices to avoid warning on reconstruction
+    G4LatticeManager::GetLatticeManager()->Reset();
+    // Clear all LogicalSurfaces
+    // NOTE: No need to redefine the G4CMPSurfaceProperties
+    G4CMPLogicalBorderSurface::CleanSurfaceTable();
+  }
+
+  Caustic_DefineMaterials();
+  Caustic_SetupGeometry();
+  fConstructed = true;
+
+  return fWorldPhys;
+}
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo....
+
+void Caustic_PhononDetectorConstruction::Caustic_DefineMaterials()
+{
+  G4NistManager* nistManager = G4NistManager::Instance();
+
+  fLiquidHelium = nistManager->FindOrBuildMaterial("G4_AIR"); // to be corrected
+  fBolometer = nistManager->FindOrBuildMaterial("G4_Al");
+
+//This lines works for Sapphire
+//
+ fOxigen = nistManager->FindOrBuildMaterial("G4_O");
+fSubstrate = new G4Material("fSubstrate", 3.98*g/cm3, 2);
+fSubstrate->AddElement(nistManager->FindOrBuildElement("Al"), 2);
+fSubstrate->AddElement(nistManager->FindOrBuildElement("O"), 3);
+//
+}
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo....
+
+void Caustic_PhononDetectorConstruction::Caustic_SetupGeometry()
+{
+    bool checkOverlaps = true;
+  //
+  // World
+  //
+  G4VSolid* worldSolid = new G4Box("World",6.*cm,6.*cm,6.*cm);
+  G4LogicalVolume* worldLogical =
+    new G4LogicalVolume(worldSolid,fLiquidHelium,"World");
+  worldLogical->SetUserLimits(new G4UserLimits(10*mm, DBL_MAX, DBL_MAX, 0, 0));
+  fWorldPhys = new G4PVPlacement(0,G4ThreeVector(),worldLogical,"World",0,
+                                 false,0);
+
+// Substrate
+ G4VSolid* fSubstrateSolid= new G4Box("fSubstrateSolid",0.2*cm,0.2*cm,0.2*cm);
```

Comment:
I would encourage you to use use human-readable and descriptive names for the volumes, rather than the names of internal variables.